src/database/entities/sp-performance-all-time.entity.ts

```diff
@@ -56,19 +56,67 @@ import { ServiceType } from "../types.js";
         -- Retrieval throughput (all time)
         ROUND(AVG(r.throughput_bps) FILTER (WHERE r.throughput_bps IS NOT NULL AND r.service_type = '${ServiceType.DIRECT_SP}'))::bigint as avg_throughput_bps,
         
+        -- IPFS retrieval metrics (all time - IPFS_PIN only)
+        COUNT(DISTINCT r_ipfs.id) FILTER (WHERE r_ipfs.service_type = '${ServiceType.IPFS_PIN}') as total_ipfs_retrievals,
+        COUNT(DISTINCT r_ipfs.id) FILTER (WHERE r_ipfs.status = 'success' AND r_ipfs.service_type = '${ServiceType.IPFS_PIN}') as successful_ipfs_retrievals,
+        COUNT(DISTINCT r_ipfs.id) FILTER (WHERE r_ipfs.status = 'failed' AND r_ipfs.service_type = '${ServiceType.IPFS_PIN}') as failed_ipfs_retrievals,
+        
+        -- IPFS retrieval success rate
```

Comment:
nit (not that you need to change) , but my understanding is that "success rates" are between 0 and 1 while "success percentages" are between 0 and 100.  In this case we are calling a "success percentage" a "success rate".